### DISCUSSION TITLE
useParams and useQuery hooks

### DISCUSSION THREAD

**Discussion by saidinusah** [nuxt/nuxt#24153] [Ideas]
Hello @danielroe , are there any plans for these hooks (useParams and useQuery)? If yes, i'm very interested in contributing to this ASAP

**Comment by danielroe** [comment]
Could you explain more about what they would do?

**Reply by saidinusah** [reply]
I'm looking at useParams returning all params of a route and use query doing same for route query params. Main reason for this is most devs(or only me) have to call useRoute before getting the params or query params. What do you think?

**Reply by saidinusah** [reply]
@danielroe
